### PR TITLE
feat: <textarea> form element implementation

### LINKED ISSUE #283: Enable users to configure multiline text inputs
__Is your feature request related to a problem? Please describe__

As part of our initiative to enable more use cases in form-js, we want to allow users to enter larger chunks of text via a multi-line text component.

__Describe the solution you'd like__

There are two main ways we can consider implementing this:

1. simply make the regular text-field multi-line capable via configuration 
2. create an entirely new component for multi-line inputs

Solution 1 is the fastest way to do this, the downside relative to solution 2 would be that multi-line is a different html component, and so it might make sense to have that difference represented in the palette as well. It also make the base component for text a tiny bit more complex, but honestly it's minor.

I would prefer solution 1, it avoids quite a bit of overhead and I also think it makes sense for a form developer.

__Additional context__

[Product hub issue](https://github.com/camunda/product-hub/issues/97)

### PR DESCRIPTION
**Pull request by Skaiir**
Closes #283

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1199,18 +1199,6 @@
         "js-yaml": "bin/js-yaml.js"
       }
     },
-    "node_modules/@eslint/eslintrc/node_modules/minimatch": {
```
Comment (pinussilvestrus) [inline package-lock.json:1202]: How are these changes related? Do we have any dependency updates?

File: packages/form-js-editor/src/features/palette/components/Palette.js [hunk 2]
```diff
@@ -5,6 +5,10 @@ const types = [
     label: 'Text Field',
     type: 'textfield'
   },
+  {
+    label: 'Text Area',
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/palette/components/Palette.js:9]: Maybe we should update our field labels to [use sentence case](https://camunda.com/brand/writing-style-guide/).

> Text field
> Text area

File: packages/form-js-editor/src/features/properties-panel/Util.js [hunk 3]
```diff
@@ -41,7 +41,8 @@ export const INPUTS = [
   'radio',
   'select',
   'taglist',
-  'textfield'
+  'textfield',
+  'textarea',
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/properties-panel/Util.js:45]:
```suggestion
  'textarea'
```

File: packages/form-js-editor/src/features/properties-panel/entries/DefaultValueEntry.js [hunk 4]
```diff
@@ -72,6 +74,14 @@ export default function DefaultOptionEntry(props) {
     });
   }
 
+  if(type === 'textarea') {
```
Comment (pinussilvestrus) [inline packages/form-js-editor/src/features/properties-panel/entries/DefaultValueEntry.js:77]: `npm run lint -- --fix` should do the trick here 👍

File: packages/form-js-viewer/src/render/components/form-fields/Textfield.js [hunk 5]
```diff
@@ -68,4 +68,4 @@ Textfield.type = type;
 Textfield.label = 'Text Field';
 Textfield.keyed = true;
 Textfield.emptyValue = '';
-Textfield.sanitizeValue = ({ value }) => (isArray(value) || isObject(value)) ? null : String(value);
\ No newline at end of file
+Textfield.sanitizeValue = ({ value }) => (isArray(value) || isObject(value)) ? '' : String(value);
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/render/components/form-fields/Textfield.js:71]: Can you describe why this change is needed? Do we have a test case for this?

File: packages/form-js-viewer/src/render/components/form-fields/Textarea.js [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+  const { required } = validate;
+
+  const onInput = ({ target }) => {
+
+    props.onChange({
+      field,
+      value: target.value
+    });
+
+    autoSizeTextarea(target);
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/render/components/form-fields/Textarea.js:42]: When I first type in something, the text area got's expanded immediately, although it's not needed. Not sure whether it's a [playground demo](https://demo-textarea--camunda-form-playground.netlify.app/) issue though.

Kapture 2022-11-01 at 13 33 57

File: packages/form-js-viewer/src/render/components/form-fields/Textarea.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+  };
+
+  const autoSizeTextarea = (textarea) => {
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/render/components/form-fields/Textarea.js:46]: I assume auto-resize is our general strategy here right? So we accept that it's not configurable (e.g., to have a fixed height only).

File: packages/form-js-viewer/src/render/components/form-fields/Textarea.js [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+  const { formId } = useContext(FormContext);
+
+  return <div class={ formFieldClasses(type, errors) }>
+    <Label
+      id={ prefixId(id, formId) }
+      label={ label }
+      required={ required } />
+    <textarea class="fjs-textarea"
+      disabled={ disabled }
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/src/render/components/form-fields/Textarea.js:70]: The auto resize is not working on disabled text areas (the scroll bar is visible).

image

File: packages/form-js-viewer/test/spec/render/components/form-fields/Textarea.spec.js [hunk 9]
```diff
@@ -0,0 +1,255 @@
...
+} from '../../../../TestHelper';
+
+import { WithFormContext } from './helper';
+
+const spy = sinon.spy;
+
+let container;
+
+
+describe('Textarea', function() {
```
Comment (pinussilvestrus) [inline packages/form-js-viewer/test/spec/render/components/form-fields/Textarea.spec.js:20]: Don't we have any test coverage for the auto resize behavior?

File: packages/form-js-viewer/src/render/components/form-fields/Textarea.js (hunk 8 above)
Comment (pinussilvestrus) [reply packages/form-js-viewer/src/render/components/form-fields/Textarea.js:70]: The same happens when using larger input data for a text area.

image

**Review by pinussilvestrus — CHANGES_REQUESTED** [review]
Looks good so far functionality and code wise, just some minor comments 👍 Let's also ensure we test the auto resize behavior (if possible).

File: package-lock.json (hunk 1 above)
Comment (Skaiir) [reply package-lock.json:1202]: They are not, but I did run an audit and fix as we had critical vulnerabilities.

File: packages/form-js-viewer/src/render/components/form-fields/Textarea.js (hunk 7 above)
Comment (Skaiir) [reply packages/form-js-viewer/src/render/components/form-fields/Textarea.js:46]: We can add this as a visual configuration in the future, but for the initial version of this, I don't think it's worth it.

File: package-lock.json (hunk 1 above)
Comment (pinussilvestrus) [reply package-lock.json:1202]: Ah alright. Let's mark this in an additional commit.

File: packages/form-js-viewer/src/render/components/form-fields/Textfield.js (hunk 5 above)
Comment (Skaiir) [reply packages/form-js-viewer/src/render/components/form-fields/Textfield.js:71]: As I built out the `textArea`, I noticed that I was sanitizing the `textField` to a null value. This is however incorrect as we use empty string as our empty state for `textField`, never null. 

This addresses it and yes we do have a test case for it: https://github.com/bpmn-io/form-js/blob/183e8cd911da92c9bafe38e0206e06e27b8d03cc/packages/form-js-viewer/test/spec/import/Importer.spec.js#L326

File: packages/form-js-viewer/src/render/components/form-fields/Textarea.js (hunk 6 above)
Comment (Skaiir) [reply packages/form-js-viewer/src/render/components/form-fields/Textarea.js]: Behavior is now very consistent

File: packages/form-js-viewer/test/spec/render/components/form-fields/Textarea.spec.js (hunk 9 above)
Comment (Skaiir) [reply packages/form-js-viewer/test/spec/render/components/form-fields/Textarea.spec.js:20]: There now is.

**Review by pinussilvestrus — APPROVED** [review]
Looks good to me 🎉

**Comment by pinussilvestrus** [comment]
For some reason firefox is broken: https://github.com/bpmn-io/form-js/actions/runs/3387857007/jobs/5638190490#step:6:128

I will investigate whether this is related to this PR or not.

**Comment by pinussilvestrus** [comment]
It's not related to this branch, so let's merge and fix at another place.
